### PR TITLE
[Nextjs] Create plugins approach for the nextjs middleware

### PR DESCRIPTION
**Pull request by illiakovalenko**
## Description / Motivation
* Each middleware should return `response` in order to use it by other middlewares
* Each middleware should have `order` to be defined

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by sc-addypathania** [comment]
Looks good but I guess part of this feature was also to break personalize add related stuff present in `packages\create-sitecore-jss\src\templates\nextjs-personalize\src\lib\page-props-factory` into its own plugin.

**Review by sc-addypathania — CHANGES_REQUESTED** [review]
Looks good but I guess part of this feature was also to break personalize related stuff present in `packages\create-sitecore-jss\src\templates\nextjs-personalize\src\lib\page-props-factory` into its own plugin.

**Comment by sc-addypathania** [comment]
@illiakovalenko I noticed that we have a separate PBI to address the changes I asked [here](https://dev.azure.com/Sitecore-PD/Products/_sprints/backlog/JSS/Products/PI-work/PI30/PI30-Sprint4-Mon?workitem=522572)
So we don't need to address that here.
Approving it.

**Review by sc-addypathania — APPROVED** [review]

File: packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+   * Detect order when the plugin should be called, e.g. 0 - will be called first (can be a plugin which data is required for other plugins)
+   */
+  order: number;
+  /**
+   * A middleware to be called, it's required to return @type {NextResponse} for other middlewares
+   */
+  (req: NextRequest, res: NextResponse): Promise<NextResponse>;
+}
+
+export default async function middleware(req: NextRequest): Promise<NextResponse> {
```
Comment (ambrauer) [inline packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts:16]: Should we add the `NextFetchEvent` parameter and also pass along to each plugin?

**Review by ambrauer — APPROVED** [review]
Just one suggestion/question, otherwise looks good!

File: packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/create-sitecore-jss/src/templates/nextjs/src/lib/middleware/index.ts]: Done

**Pull request merged**